python-lib/xlsx_writer.py

```diff
@@ -104,33 +106,94 @@ def auto_size_column_width(worksheet: Worksheet):
     worksheet.column_dimensions = dimension_holder
 
 
-def dataframes_to_xlsx(input_dataframes_names, xlsx_abs_path, dataframe_provider):
+
+
+# code inspired from https://openpyxl.readthedocs.io/en/stable/_modules/openpyxl/worksheet/copier.html
+def copy_sheet_to_workbook(source_sheet: Worksheet, target_workbook: Workbook) -> Worksheet:
     """
-    Write the input datasets into same excel into the folder
-    :param input_datasets_names:
-    :param writer:
-    :return:
+    Copy the source worksheet as a new worksheet in the target workbook
+    :param source_sheet: the source sheet
+    :param target_workbook: the workbook used to store the new sheet
+    :return: a reference to the created sheet inside the workbook
+    """
+    logger.info(f"Copying sheet {source_sheet.title} to target workbook")
+    target_sheet = target_workbook.create_sheet(source_sheet.title)
+    for row in source_sheet:
+        for cell in row:
+            new_cell = target_sheet.cell(row=cell.row, column=cell.column, value=cell.value)
+            new_cell.data_type = cell.data_type
+            if cell.has_style:
+                new_cell.font = copy(cell.font)
+                new_cell.border = copy(cell.border)
+                new_cell.fill = copy(cell.fill)
+                new_cell.number_format = copy(cell.number_format)
+                new_cell.alignment = copy(cell.alignment)
+
+    return target_sheet
+
+def rename_too_long_dataset_names(input_dataset_names: List[str]) -> Dict[str, str]:
+    """
+    Excel allows for only maximum 30 chars in the sheet names, so if some DS have more than 30 chars :
+        - truncate the name to 28 chars
+        - Add an index from 00 to 99 at the end in case of overlap
+    :param input_dataset_names: the list of dataset names to remap
+    :returns: a Dict[str, str] mapping the DS names with the sheet names
     """
-    logger.info("Writing output xlsx file ...")
-    writer = pd.ExcelWriter(xlsx_abs_path, engine='openpyxl')
 
-    for name in input_dataframes_names:
-        df = dataframe_provider(name)
+    return_map = {}
+    index_rename=-1
```

Comment:
I think it would be a little bit cleaner to have it start at 0 and increment at the end of the if condition

Reply:
I did that first, but I find code less readable or easier to break, because one can miss to increment the index at the end of the condition, that's why I like to start with -1, I've saw this trick on a forum once

Reply:
makes sense